Automattic/jetpack
# SEO: add a per post/page HTML title option

This PR adds an option to override the HTML `<title>` tag at the individual post/ page level.

#### Changes proposed in this Pull Request:

* This PR adds a new meta field in the SEO module  that allows a custom value for the `<title>` tag to be specified at the post level.
* It then handles the output of the custom `<title>` value when specified for an individual page/ post.

#### Jetpack product discussion


pbNhbs-4eC-p2

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Check out this branch on your test site
* Before beginning the test, make sure the SEO module is enabled by navigating to "Jetpack > Settings" and searching for "SEO". Turn this module on by clicking on the toggle:
<img width="1015" alt="Screen Shot 2022-11-02 at 3 57 02 PM" src="https://user-images.githubusercontent.com/18016357/199610413-9e479a8a-4626-4489-8826-1d3fb1d4d004.png">

* Navigate to an individual post, open the front-end output for the post in a new tab and take note of the page title displaying in the HTML/ browser tab. This should be the default HTML title.
* In the post editor, click on the Jetpack icon in the upper right-hand corner, this will show Jetpack-specific settings in the post editor sidebar.
<img width="294" alt="Screen Shot 2022-11-02 at 3 57 39 PM" src="https://user-images.githubusercontent.com/18016357/199610487-69b82b12-48a8-4d4d-8999-0711c6b78a59.png">

* Under "SEO title", set a custom title
* Click the "Update" button to update your new custom meta value
* Refresh the post on the front-end of the site and confirm that the `<title>` tag and browser tab now display your custom title.

* If you have the Jetpack social connections module enabled, you should also see the `og:title` meta tag updated to contain your custom title string.
* This PR makes some small modifications to how the custom SEO description is fetched, please check that adding custom SEO descriptions still work as expected for posts and pages.
* Change post status to "Draft"
* On the Edit page of the post, click the "Publish" button
* The sidebar should be presented with both SEO Page Title and SEO Description fields:
<img width="283" alt="CleanShot 2022-11-11 at 16 48 50@2x" src="https://user-images.githubusercontent.com/8419292/201441175-dbe1f950-69c3-416f-9afe-75d5074db060.png">
